ambry-store/src/main/java/com.github.ambry.store/PersistentIndex.java

```diff
@@ -576,31 +581,11 @@ private boolean isExpired(IndexValue value) {
   FindInfo findEntriesSince(FindToken token, long maxTotalSizeOfEntries) throws StoreException {
     long startTimeInMs = time.milliseconds();
     try {
-      boolean tokenWasReset = false;
+      boolean tokenWasReset;
       Offset logEndOffsetBeforeFind = log.getEndOffset();
-      StoreFindToken storeToken = (StoreFindToken) token;
-      // validate token
-      if (storeToken.getSessionId() == null || storeToken.getSessionId().compareTo(sessionId) != 0) {
-        // the session has changed. check if we had an unclean shutdown on startup
-        if (!cleanShutdown) {
-          // if we had an unclean shutdown and the token offset is larger than the logEndOffsetOnStartup
-          // we reset the token to logEndOffsetOnStartup
-          if (!storeToken.getType().equals(StoreFindToken.Type.Uninitialized)
-              && storeToken.getOffset().compareTo(logEndOffsetOnStartup) > 0) {
-            logger.info("Index : " + dataDir + " resetting offset after not clean shutdown " + logEndOffsetOnStartup
-                + " before offset " + storeToken.getOffset());
-            storeToken = new StoreFindToken(logEndOffsetOnStartup, sessionId);
-            tokenWasReset = true;
-          }
-        } else if (!storeToken.getType().equals(StoreFindToken.Type.Uninitialized)
-            && storeToken.getOffset().compareTo(logEndOffsetOnStartup) > 0) {
-          logger.error(
-              "Index : " + dataDir + " invalid token. Provided offset is outside the log range after clean shutdown");
-          // if the shutdown was clean, the offset should always be lesser or equal to the logEndOffsetOnStartup
-          throw new IllegalArgumentException(
-              "Invalid token. Provided offset is outside the log range after clean shutdown");
-        }
-      }
+      AtomicReference<StoreFindToken> storeFindTokenAtomicReference = new AtomicReference<>((StoreFindToken) token);
```

Comment:
Yeah, it makes sense to have the method return the new token - and also name the method accordingly (as in @vgkholla's comment).
It also looks to me that the inclusive/exclusive aspect should really go inside the token. The token should say *exactly* where to start reading from. Is that something you guys are comfortable changing now (since we are updating the version)?